postgres/postgres.go

```diff
@@ -465,26 +493,71 @@ func (ds Datastore) UpdateFilter(isAuthenticated bool, filter *models.Filter) er
 		return err
 	}
 
-	updateStatement, err := updateStatement(filter)
-	if err != nil {
-		return err
+	rowsAffected := int64(0)
+
+	if filter.State != "" {
+		results, err := tx.Stmt(ds.updateFilterState).Exec(filter.FilterID, filter.State)
```

Comment:
Can we test that the expected query is executed in a transaction if the filter.State != ""